twyla/service/event_bus.py

```diff
@@ -16,13 +21,49 @@ def __init__(self, callback):
 class EventBus:
 
     def __init__(self, config_prefix: str):
+        self.config_prefix = config_prefix
+        self.event_listeners = {}
         self.queue_manager = queues.QueueManager(config_prefix)
 
-    async def listen(self, event_name: str, event_group: str, callback):
+
+    def listen(self, event_name: str, event_group: str, callback):
+        self.event_listeners[event_name] = (callback, event_group)
+
+
+    async def start(self):
         await self.queue_manager.connect()
-        await self.queue_manager.listen(event_name, event_group, MessageToEventAdapter(callback))
+        for event_name, (callback, group) in self.event_listeners.items():
+            await self.queue_manager.listen(event_name, group, MessageToEventAdapter(callback))
+
 
     async def emit(self, event):
         await self.queue_manager.connect()
         data = event.to_json()
         await self.queue_manager.emit(event.event_name, data)
+
+
+    async def main_task(self, aio_loop):
```

Comment:
I'm confused--because I still don't completely understand how `asyncio` works--as to why we are passing the `aio_loop` that creates the task for this `main_task`. Are we passing this `aio_loop` so that we can schedule new coroutines?

Reply:
(Caveat: asyncio is not 100% clear for me, either. Need some reading up to do)

There are two ways to get a grip on the asyncio loop; either call `asyncio.get_event_loop`, or pass it around. The former creates a new loop and returns that in case the old one is closed and removed, so it is common practice to pass it around. But as mentioned, I'm not expert. I will probably need to read a couple of things in order to clean up some non-standard usage, also for the stuff in the upcoming PR.

Reply:
Ah ok, that makes sense. Thanks!